digitalocean/resource_digitalocean_project_resource_test.go

```diff
@@ -0,0 +1,104 @@
+package digitalocean
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccDigitalOceanProjectResource_Basic(t *testing.T) {
+	projectName := generateProjectName()
+	dropletName := generateDropletName()
+
+	config := fmt.Sprintf(`
+resource "digitalocean_project" "foo" {
+  name = "%s"
+}
+
+resource "digitalocean_droplet" "foobar" {
+  name      = "%s"
+  size      = "512mb"
+  image     = "centos-7-x64"
+  region    = "nyc3"
+  user_data = "foobar"
+}
+
+resource "digitalocean_project_resource" "barfoo" {
+  project = digitalocean_project.foo.id
+  resource = digitalocean_droplet.foobar.urn
+}
+`, projectName, dropletName)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDigitalOceanProjectResourceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testResourceInstanceState("digitalocean_project_resource.barfoo", testAccCheckDigitalOceanProjectResourceExists),
+				),
+				ExpectNonEmptyPlan: true,
```

Comment:
> The plan was non-empty on destroy after running the test for some reason, maybe destruction of the droplet?

The diff is cause by the `digitalocean_project` resource now having a Droplet in its resources attribute that doesn't match the config.

I think what we want here is to is to set the `resources` attribute of the `digitalocean_project` to be both:

```
Optional: true,
Computed: true,
```

https://github.com/terraform-providers/terraform-provider-digitalocean/blob/81eb1686779b1cdfa876f71797a56f21f2a241e6/digitalocean/resource_digitalocean_project.go#L81-L86

While the two methods of adding a resource to a project will still be mutually exclusive, it will allow for using `digitalocean_project` without resources together with `digitalocean_project_resource` This is similar to how we've handled the `digitalocean_volume_attachment` and the `digitalocean_droplet` `volume_ids` attribute.

Reply:
I'll try that fix.

Reply:
Setting `Computed: true` worked to fix the test for `digitalocean_project_resource`. I didn't run the tests for `digitalocean_project` due to cost though.